src/routes/v1/students.route.js

```diff
@@ -0,0 +1,184 @@
+const express = require('express');
+const validate = require('../../middlewares/validate');
+const userValidation = require('../../validations/user.validation');
+const studentsController = require('../../controllers/students.controller');
+const auth = require('../../middlewares/auth');
+
+
+const router = express.Router();
+router.route('/students').post(auth('teacher'), validate(userValidation.createUser), studentsController.createStudent);
+router.route('/students/:studentsId').get(auth('teacher'), validate(userValidation.getUser), studentsController.getStudent);
+router.route('/students/:studentsId').get(auth('student'), validate(userValidation.getUser), studentsController.getStudent);
+router.route('/students/:studentsId').patch(auth('teacher'), validate(userValidation.updateUser), studentsController.updateStudent);
+router.route('/students/:studentsId').patch(auth('student'), validate(userValidation.updateUser), studentsController.updateStudent);
```

Comment:
If you have the same path it will always match the first function and then fail, if you are a student. You should use `auth()` so both stundents and teachers can use the endpoint